src/Raven.Server/Documents/Sharding/Handlers/Processors/Smuggler/ShardedSmugglerHandlerProcessorForExport.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations;
+using Raven.Server.Documents.Handlers.Processors.Smuggler;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Smuggler.Documents.Data;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Smuggler
+{
+    internal class ShardedSmugglerHandlerProcessorForExport : AbstractSmugglerHandlerProcessorForExport<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedSmugglerHandlerProcessorForExport([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask ExportAsync(JsonOperationContext context, long? operationId)
+        {
+            if(operationId.HasValue == false)
+                operationId = RequestHandler.DatabaseContext.Operations.GetNextOperationId();
+
+            await RequestHandler.Export(context, RequestHandler.DatabaseContext.DatabaseName, ExportShardedDatabaseInternalAsync, RequestHandler.DatabaseContext.Operations, operationId.Value);
```

Comment:
Export method should not be in RequestHandler

Reply:
moved to an abstract smuggler processor